pixl_imaging/src/pixl_imaging/_processing.py

```diff
@@ -143,27 +145,122 @@ async def _add_project_to_study(
+        raise PixlDiscardError(msg)
+
+    if _is_daytime() or _is_weekend():
+        msg = (
+            f"Failed to find study {study.message.study_uid} in primary archive. "
+            "Not querying secondary archive during the daytime or on the weekend."
         )
+        raise PixlDiscardError(msg)
+
+    logger.info(
+        "Failed to find study {} in primary archive, trying secondary archive",
+        study.message.study_uid,
+    )
+    query_id = await _find_study_in_archive(
+        study=study,
+        orthanc_raw=orthanc_raw,
+        modality=config("SECONDARY_DICOM_MODALITY"),
+    )
+
     if query_id is None:
-        msg = "Failed to find in the VNA"
+        msg = f"Failed to find study {study.message.study_uid} in primary or secondary archive."
+        raise PixlDiscardError(msg)
+
+    # Check the study is in the online secondary archive
+    query_answers = await orthanc_raw.get_remote_query_answers(query_id)
+    query_answer_content = await orthanc_raw.get_remote_query_answer_content(
+        query_id=query_id,
+        answer_id=query_answers[0],
+    )
+    availablility_tag = "0008,0056"
+    availability = query_answer_content[availablility_tag]["Value"]
+    if availability != "ONLINE":
+        msg = (
+            f"Study {study.message.study_uid} found in {availability} secondary archive "
+            "but we only pull from the online secondary archive."
+        )
         raise PixlDiscardError(msg)
+
     return query_id
 
 
+async def _find_study_in_archive(
+    orthanc_raw: Orthanc,
+    study: ImagingStudy,
+    modality: str,
+) -> Optional[str]:
+    """
+    Query the primary archive for the study using its UID.
+    If UID is not available, query on MRN and accession number.
+    """
+    # We don't want to normalize the query otherwise only MainDicomTags will be returned
+    # (InstanceAvailability will be ignored)
+    additional_data = {"Normalize": False}
+    query_response = await orthanc_raw.query_remote(
+        data=study.orthanc_uid_query_dict | additional_data,
+        modality=modality,
+    )
+
+    if query_response is not None:
+        return query_response
+
+    logger.info(
+        "No study found in modality {} with UID {}, trying MRN and accession number",
+        modality,
+        study.message.study_uid,
+    )
+    return await orthanc_raw.query_remote(
+        study.orthanc_query_dict | additional_data,
+        modality=modality,
+    )
+
+
+def _is_daytime() -> bool:
+    """Check if the current time is between 8 am and 8 pm."""
+    timezone = ZoneInfo(config("TZ", default="Europe/London"))
+    after_8am = datetime.time(8, 00) <= datetime.datetime.now(tz=timezone).time()
+    before_8pm = datetime.datetime.now(tz=timezone).time() <= datetime.time(20, 00)
+    return after_8am and before_8pm
+
+
+def _is_weekend() -> bool:
+    """Check if it's the weekend."""
+    timezone = ZoneInfo(config("TZ", default="Europe/London"))
```

Comment:
I guess we should think about where we want to define the defaults. Currently we've got them in the docker envs and in the code. Don't particularly mind where but the default should only be in one place IMHO

Reply:
yeah that would be much neater. I guess it makes sense to set it in docker

Reply:
I've removed the defaults here and only set them in the docker compose configs